model/histogram/validate.go

```diff
@@ -0,0 +1,123 @@
+package histogram
+
+import (
+	"fmt"
+	"math"
+
+	"github.com/pkg/errors"
+)
+
+var (
+	ErrHistogramCountNotBigEnough    = errors.New("histogram's observation count should be at least the number of observations found in the buckets")
+	ErrHistogramCountMismatch        = errors.New("histogram's observation count should equal the number of observations found in the buckets (in absence of NaN)")
+	ErrHistogramNegativeBucketCount  = errors.New("histogram has a bucket whose observation count is negative")
+	ErrHistogramSpanNegativeOffset   = errors.New("histogram has a span whose offset is negative")
+	ErrHistogramSpansBucketsMismatch = errors.New("histogram spans specify different number of buckets than provided")
+)
+
+func ValidateHistogram(h *Histogram) error {
```

Comment:
Moved from tsdb/head_append.go

Reply:
Maybe this could be a method of `Histogram` now, just called `Validate`?

Would also be good to give it a doc comment to explain what is validated.

Reply:
Sure, makes sense. Changed from `func ValidateHistogram(h *Histogram) error` to `func (h *Histogram) Validate() error` and added doc comments.